model/textparse/openmetricsparse.go

```diff
@@ -222,9 +222,82 @@ func (p *OpenMetricsParser) Exemplar(e *exemplar.Exemplar) bool {
 // CreatedTimestamp returns nil as it's not implemented yet.
 // TODO(bwplotka): https://github.com/prometheus/prometheus/issues/12980
 func (p *OpenMetricsParser) CreatedTimestamp() *int64 {
+	newParser := deepCopyParser(p)
+	switch t, _ := newParser.Next(); t {
+		case EntrySeries:
+			for {
+				// Check _created suffix
+				var lbs labels.Labels
+				newParser.Metric(&lbs)
+				name := lbs.Get(model.MetricNameLabel)
+				if name[len(name)-8:] != "_created" {
```

Comment:
nit: To avoid that magic number 8, one thing you can do is store "_created" as a variable and use the len of it.

Or maybe we should just trim everything before the last _ that way we can handle any suffix. The only edge case here is if there is no _, e.g. just foo and not foo_total.

Reply:
covered via our pair programming session